src/blocks/blockCTATransitions.ts

```diff
@@ -11,19 +12,47 @@ export const blockCTATransitions = base.createBlock({
 	produce({ options }) {
 		return {
 			addons: [
-				blockGitHubActionsCI({
-					jobs: [
-						{
-							checkoutWith: {
-								"fetch-depth": "0",
-								ref: "${{github.event.pull_request.head.ref}}",
-								repository:
-									"${{github.event.pull_request.head.repo.full_name}}",
-								token: '"${{ secrets.ACCESS_TOKEN }}"',
-							},
-							if: "${{ startsWith(github.head_ref, 'renovate/') && contains(github.event.pull_request.title, 'create-typescript-app') }}",
+				blockPackageJson({
+					properties: {
+						devDependencies: {
+							"create-typescript-app": packageData.version,
+						},
+					},
+				}),
+				blockRepositoryBranchRuleset({
+					requiredStatusChecks: ["CTA Transitions"],
+				}),
+			],
+			files: {
+				".github": {
+					workflows: {
+						"cta-transitions.yml": createSoloWorkflowFile({
+							if: `(github.actor == '${options.owner}' || github.actor == 'renovate[bot]') && startsWith(github.head_ref, 'renovate/') && contains(github.event.pull_request.title, 'create-typescript-app')`,
```

Comment:
TODO: I'm under the impression _skipped_ jobs still cause rulesets to prevent PRs from being merged, not just _failed_. Need to verify...